### ISSUE TITLE
Search for better and more exclusive macros. Possibly configure CMake to avoid include location variance.

### ISSUE THREAD

**Issue by harrisi** [harrisi/Console#2]

**Comment by harrisi** [comment]
I don't yet have an answer to this, but I do know that the current platform checks are not very complete and make a fair amount of assumptions. This could definitely be improved, but I'm also not too concerned about it for now. It currently supports Windows, Linux (at least a couple distros), and macOS Sierra just fine.

**Comment by R030t1** [comment]
The platform with the most issues is Windows. Editing the CMakeFiles to search the included dependency repositories last would probably work.

On Unix-like platforms it's possible to avoid running `git submodule init` and `git submodule update` to use the system-wide libraries, but with Windows, there is no default install location to search. FreeType may have an old installer on the internet somewhere but SDL2 has since dropped support completely for use in this way.

**Comment by harrisi** [comment]
"but with Windows, there is no default install location to search." That's not really an issue though, is it? In an actual use of this, SDL2.dll would be shipped with the executable. Either that or build it on the client machine, but that seems silly. The main issue with trying to find it somewhere is that most machines probably don't have it anywhere anyway.

**Comment by R030t1** [comment]
Yes, it's customary to ship libraries with executables in Windows (and in MacOS when developing an .app). There was a point in time where some common libraries provided installers that placed the DLLs into System32 or some equivalent to avoid this hassle.

What this means is that we are here, having this discussion, realizing the necessity of patching or replacing the default CMake files so that they can search the project's root directory.
